crates/ere-nexus/src/lib.rs

```diff
@@ -0,0 +1,208 @@
+use std::path::PathBuf;
+use std::time::Instant;
+
+use nexus_sdk::compile::cargo::CargoPackager;
+use nexus_sdk::compile::{Compile, Compiler as NexusCompiler};
+use nexus_sdk::stwo::seq::{Proof, Stwo};
+use nexus_sdk::{Local, Prover, Verifiable};
+use tracing::info;
+use zkvm_interface::{
+    Compiler, Input, ProgramExecutionReport, ProgramProvingReport, ProverResourceType, zkVM,
+    zkVMError,
+};
+
+mod error;
+use crate::error::ProveError;
+use error::{CompileError, NexusError, VerifyError};
+
+/// NOTE: nexus's guest package name must be "ere-nexus-guest"
+/// TODO: This should be configurable in the future, but for now we hardcode it.
+const PACKAGE: &str = "ere-nexus-guest";
+
+#[allow(non_camel_case_types)]
+pub struct NEXUS_TARGET;
+
+impl Compiler for NEXUS_TARGET {
+    type Error = NexusError;
+
+    type Program = PathBuf;
+
+    fn compile(_: &std::path::Path) -> Result<Self::Program, Self::Error> {
+        let mut prover_compiler = NexusCompiler::<CargoPackager>::new(PACKAGE);
+        let elf_path = prover_compiler
+            .build()
+            .map_err(|e| CompileError::Client(e.into()))?;
+
+        Ok(elf_path)
+    }
+}
+
+pub struct EreNexus {
+    program: <NEXUS_TARGET as Compiler>::Program,
+}
+
+impl EreNexus {
+    pub fn new(
+        program: <NEXUS_TARGET as Compiler>::Program,
+        _resource_type: ProverResourceType,
+    ) -> Self {
+        Self { program }
+    }
+}
+impl zkVM for EreNexus {
+    fn execute(&self, inputs: &Input) -> Result<zkvm_interface::ProgramExecutionReport, zkVMError> {
+        let start = Instant::now();
+
+        // let mut public_input = vec![];
+        let mut private_input = vec![];
+        for input in inputs.iter() {
+            private_input.extend(
+                input
+                    .as_bytes()
+                    .map_err(|e| NexusError::Prove(ProveError::Client(e.into())))
+                    .map_err(zkVMError::from)?,
+            );
+        }
+
```

Comment:
I think we missing `prover.run_with_input` here

Reply:
Could you elaborate more what's that mean? You mean it doesn't count the execution cycle? If so we could leave it as is.